analyzer.py

```diff
@@ -0,0 +1,277 @@
+        bodyparts2 = self._data_processor.get_bodyparts_from_xma(trial2_path, mode='rgb')
+        markers_in_common = [marker for marker in bodyparts1 if marker in bodyparts2]
+
+        trial1_csv = pd.read_csv(os.path.join(trial1_path, f'{trial1}.csv'))
+        trial2_csv = pd.read_csv(os.path.join(trial2_path, f'{trial2}.csv'))
+
+        avg_distances = []
+        for marker in markers_in_common:
+            avg_x1, avg_y1 = trial1_csv[f'{marker}_X'].mean(), trial1_csv[f'{marker}_Y'].mean()
+            avg_x2, avg_y2 = trial2_csv[f'{marker}_X'].mean(), trial2_csv[f'{marker}_Y'].mean()
+  
+            # Calculate the distance between the average positions for this marker in the two trials
+            distance = math.sqrt((avg_x2 - avg_x1) ** 2 + (avg_y2 - avg_y1) ** 2)
+            avg_distances.append(distance)
+
+        # Calculate the mean of the distances to get an overall similarity measure
+        marker_similarity = sum(avg_distances) / len(avg_distances) if avg_distances else 0
+
+        return marker_similarity
+
+    def get_max_dissimilarity_for_trial(self, trial_path, window):
+        '''Calculate the dissimilarity within the trial given the frame sliding window.'''        
+        video1 = cv2.VideoCapture(self._data_processor.find_cam_file(trial_path, 'cam1'))
+        video2 = cv2.VideoCapture(self._data_processor.find_cam_file(trial_path, 'cam2'))
+
+        hashes1 = self._hash_trial_video(video1)
+        hashes2 = self._hash_trial_video(video2)
+
+        return self._find_dissimilar_regions(hashes1, hashes2, window)
+
+    def _split_dlc_to_xma(self, trial, save_hdf=True):
+        '''Takes the output from RGB deeplabcut and splits it into XMAlab-readable output'''
+        bodyparts_xy = []
+        yaml = YAML()
+        with open(self._dlc_config) as dlc_config:
+            dlc = yaml.load(dlc_config)
+        iteration = dlc['iteration']
+        trial_path = os.path.join(self._trials_path, trial)
+
+        rgb_parts = self._data_processor.get_bodyparts_from_xma(trial_path, mode='rgb')
+        for part in rgb_parts:
+            bodyparts_xy.append(part+'_X')
+            bodyparts_xy.append(part+'_Y')
+
+        csv_path = [file for file in os.listdir(f'{trial_path}/it{iteration}') if '.csv' in file and '-2DPoints' not in file]
+        if len(csv_path) > 1:
+            raise FileExistsError('Found more than 1 data CSV for RGB trial. Please remove CSVs from older analyses from this folder before analyzing.')
+        if len(csv_path) < 1:
+            raise FileNotFoundError(f'Couldn\'t find data CSV for trial {trial}. Something wrong with DeepLabCut?')
+
+        csv_path = csv_path[0]
+        xma_csv_path = f'{trial_path}/it{iteration}/{trial}-Predicted2DPoints.csv'
+
+        df = pd.read_csv(f'{trial_path}/it{iteration}/{csv_path}', skiprows=1)
+        df.index = df['bodyparts']
+        df = df.drop(columns=df.columns[[df.loc['coords'] == 'likelihood']])
```

Comment:
This is an error that was removed in a branch of mine... you only need one set of brackets here, using boolean logic with the DataFrame to drop columns with the name "likelihood"

Reply:
done